mne/time_frequency/csd.py

```diff
@@ -1,63 +1,522 @@
+#          Roman Goj <roman.goj@gmail.com>
 #
 # License: BSD (3-clause)
 
 import copy as cp
+import numbers
 
 import numpy as np
-
-from ..io.pick import pick_types
-from ..utils import logger, verbose, warn
+from .tfr import cwt, morlet
+from ..io.pick import pick_types, pick_channels
+from ..utils import logger, verbose, warn, copy_function_doc_to_method_doc
+from ..viz.misc import plot_csd
 from ..time_frequency.multitaper import (dpss_windows, _mt_spectra,
                                          _csd_from_mt, _psd_from_mt_adaptive)
-
-from ..externals.six.moves import xrange as range
+from ..parallel import parallel_func
+from ..externals.h5io import read_hdf5, write_hdf5
 
 
 class CrossSpectralDensity(object):
     """Cross-spectral density.
 
+    Given a list of time series, the CSD matrix denotes for each pair of time
+    series, the cross-spectral density. This matrix is symmetric and internally
+    stored as a vector.
+
+    This object can store multiple CSD matrices: one for each frequency.
+    Use ``.get_matrix(freq)`` to obtain an CSD matrix as an ndarray.
+
     Parameters
     ----------
-    data : array of shape (n_channels, n_channels)
-        The cross-spectral density matrix.
-    ch_names : list of string
-        List of channels' names.
-    projs :
-        List of projectors used in CSD calculation.
-    bads :
-        List of bad channels.
-    freqs : float | list of float
-        Frequency or frequencies for which the CSD matrix was calculated. If a
-        list is passed, data is a sum across CSD matrices for all frequencies.
+    data : ndarray, shape ((n_series**2 + n_series) / 2, n_frequencies)
+        For each frequency, the cross-spectral density matrix in vector format.
+    names : list of string
+        List of string descriptions for each time series (e.g. channel names)
+    tmin : float
+        Start of the time window for which CSD was calculated in seconds.
+    tmax : float
+        End of the time window for which CSD was calculated in seconds.
+    frequencies : float | list of float | list of list of float
+        Frequency or frequencies for which the CSD matrix was calculated. When
+        averaging across frequencies (see the :func:`CrossSpectralDensity.mean`
+        function), this will be a list of lists that contains for each
+        frequency bin, the frequencies that were averaged. Frequencies should
+        always be sorted.
     n_fft : int
-        Length of the FFT used when calculating the CSD matrix.
+        The number of FFT points or samples that have been used in the
+        computation of this CSD.
+    projs : list of Projection | None
+        List of projectors to apply to timeseries data when using this CSD
+        object to compute a DICS beamformer. Defaults to None, which means no
+        projectors will be applied.
+    is_sum : bool
+        Whether the CSD matrix represents a sum (or average) across
+        frequencies. Defaults to False.
```

Comment:
isn't it redundant with the fact that frequencies is a list of list?

Reply:
Hrm, yes it is! I'll make it a property instead.